pkg/analyzer/analyzers/stripe/stripe.go

```diff
@@ -162,70 +167,87 @@ func checkValidity(cfg *config.Config, key string) (bool, error) {
 	return false, nil
 }
 
-func AnalyzePermissions(cfg *config.Config, key string) {
-
+func AnalyzePermissions(cfg *config.Config, key string) (*SecretInfo, error) {
 	// Check if secret, publishable, or restricted key
 	var keyType, keyEnv string
 	keyType, err := checkKeyType(key)
 	if err != nil {
-		color.Red("[x] ", err.Error())
-		return
-	}
-
-	if keyType == PUBLISHABLE {
-		color.Red("[x] This is a publishable Stripe key. It is not considered secret.")
-		return
+		return nil, err
 	}
 
 	// Check if live or test key
 	keyEnv, err = checkKeyEnv(key)
 	if err != nil {
-		color.Red("[x] ", err.Error())
-		return
+		return nil, err
 	}
 
 	// Check if key is valid
 	valid, err := checkValidity(cfg, key)
 	if err != nil {
-		color.Red("[x] ", err.Error())
+		return nil, err
+	}
+
+	permissions, err := getRestrictedPermissions(cfg, key)
+	if err != nil {
+		return nil, err
+	}
+	// Additional details
+	// get total customers
+	// get total charges
+
+	return &SecretInfo{
+		KeyType:     keyType,
+		KeyEnv:      keyEnv,
+		Valid:       valid,
+		Permissions: permissions,
+	}, nil
+}
+
+func AnalyzeAndPrintPermissions(cfg *config.Config, key string) {
+	// ToDo: Add in logging
+	if cfg.LoggingEnabled {
+		color.Red("[x] Logging is not supported for this analyzer.")
+		return
+	}
```

Comment:
Is this true? I thought logging was supported.

Reply:
Remove the logging condition and updated the code.